sdk/servicebus/service-bus/src/util/atomXmlHelper.ts

```diff
@@ -0,0 +1,377 @@
+  HttpOperationResponse,
+  RestError,
+  stripRequest,
+  stripResponse,
+  WebResource,
+  stringifyXML,
+  parseXML,
+  ServiceClient
+} from "@azure/core-http";
+
+import * as ServiceBusConstants from "./constants";
+import { isNode } from "./utils";
+
+/**
+ * @ignore
+ * Represents the internal ATOM XML serializer interface
+ */
+export interface AtomXmlSerializer {
+  serialize(requestBodyInJson: object): object;
+
+  deserialize(response: HttpOperationResponse): Promise<HttpOperationResponse>;
+}
+
+/**
+ * @ignore
+ * Utility to execute Atom XML operations as HTTP requests
+ * @param webResource
+ * @param serializer
+ */
+export async function executeAtomXmlOperation(
+  serviceBusAtomManagementClient: ServiceClient,
+  webResource: WebResource,
+  serializer: AtomXmlSerializer
+): Promise<HttpOperationResponse> {
+  if (webResource.body) {
+    const content: object = serializer.serialize(JSON.parse(webResource.body));
+    webResource.body = stringifyXML(content, { rootName: "entry" });
+  }
+
+  let response: HttpOperationResponse = await serviceBusAtomManagementClient.sendRequest(
+    webResource
+  );
+
+  try {
+    if (response.bodyAsText) {
+      response.parsedBody = await parseXML(response.bodyAsText, { includeRoot: true });
+    }
+  } catch (err) {
+    const error = new RestError(
+      `ResponseNotInAtomXMLFormat - ${err.message}`,
+      RestError.PARSE_ERROR,
+      response.status,
+      stripRequest(response.request),
+      stripResponse(response)
+    );
+
+    throw error;
+  }
+
+  return serializer.deserialize(response);
+}
+
+/**
+ * Serializes input information to construct the Atom XML request
+ * @param resourceName
+ * @param resource
+ * @param properties
+ * @param xmlNamespace
+ */
+export function serializeToAtomXmlRequest(
+  resourceName: string,
+  resource: any,
+  properties: string[],
+  xmlNamespace: string
+): object {
+  const content: any = {};
+  content[resourceName] = {
+    $: {
+      xmlns: xmlNamespace,
+      "xmlns:i": "http://www.w3.org/2001/XMLSchema-instance"
+    }
+  };
+
+  if (resource) {
+    // Sort properties according to what is allowed by the service
+    properties.forEach((property: string) => {
+      if (resource[property] !== undefined) {
+        content[resourceName][property] = resource[property];
+      }
+    });
+  }
+  content[CoreHttpConstants.XML_METADATA_MARKER] = { type: "application/xml" };
+  return {
+    $: {
+      xmlns: "http://www.w3.org/2005/Atom"
+    },
+    updated: new Date().toISOString(),
+    content: content
+  };
+}
+
+/**
+ * Transforms response to contain the parsed data.
+ * @param nameProperties The set of 'name' properties to be constructed on the
+ * resultant object e.g., QueueName, TopicName, etc.
+ * @param response
+ * @param shouldParseResponse
+ */
+export async function deserializeAtomXmlResponse(
+  nameProperties: string[],
+  response: HttpOperationResponse
+): Promise<HttpOperationResponse> {
```

Comment:
By having the interface and utility return a copy of the value, we don't force implementations to mutate the input object. For any reason if we don't want to mutate the object and return new one, current API would help with that. Since there doesn't seem any specific benefit for the proposal, can leave this as is.